src/schema/v2/fairs.ts

```diff
@@ -116,11 +117,53 @@ export const fairsConnection: GraphQLFieldConfig<
     near: { type: Near },
     sort: FairSorts,
     status: EventStatus,
+    term: {
+      type: GraphQLString,
+      description: "Search term to match against fair names",
+    },
   }),
   description: "A list of fairs",
-  resolve: async (_root, args, { fairsLoader }) => {
+  resolve: async (
+    _root,
+    args,
+    {
+      unauthenticatedLoaders: { fairsLoader },
+      authenticatedLoaders: { matchFairsLoader },
+    }
+  ) => {
     const { size, offset, page } = convertConnectionArgsToGravityArgs(args)
 
+    // Use matchFairsLoader when term is provided
+    if (args.term) {
+      if (!matchFairsLoader)
+        throw new Error(
+          "You need to pass a X-Access-Token header to perform this action"
+        )
+
+      const { body, headers } = await matchFairsLoader({
+        term: args.term,
+        page,
+        size,
+        total_count: true,
+      })
+
+      const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+
+      return {
```

Comment:
While you're here...this should use the `paginationResolver` helper - that's preferred and basically just wraps all this up for you.